### PR TITLE
Add type hints to test.with_dummyserver.test_chunked_transfer

### PR DESCRIPTION
**Pull request by hramezani**

### CODE REVIEW

File: test/with_dummyserver/test_chunked_transfer.py [hunk 1]
```diff
@@ -33,9 +36,9 @@ def socket_handler(listener):
 
         self._start_server(socket_handler)
 
-    def test_chunks(self):
+    def test_chunks(self) -> None:
         self.start_chunked_handler()
-        chunks = ["foo", "bar", "", "bazzzzzzzzzzzzzzzzzzzzzz"]
+        chunks = [b"foo", b"bar", b"", b"bazzzzzzzzzzzzzzzzzzzzzz"]
```
Comment (bluetech) [inline test/with_dummyserver/test_chunked_transfer.py:41]: It does seem like `request_chunked` explicitly supports `str` chunks:

https://github.com/urllib3/urllib3/blob/cff8b5ba3d2ba9338e31c102e1e91273124df297/src/urllib3/connection.py#L326-L327

I'm not sure if this is a case "we support it but don't do this, so type annotation forbid this" or "this is supported so `_TYPE_BODY` should allow it". To me chunked + str sounds like a bad idea but I don't really know.

**Review by bluetech — COMMENTED** [review]
I left a comment regarding the str -> bytes changes, other than that LGTM.

File: test/with_dummyserver/test_chunked_transfer.py (hunk 1 above)
Comment (hramezani) [reply test/with_dummyserver/test_chunked_transfer.py:41]: @sethmlarson @pquentin Could you please elaborate on this?

Comment (pquentin) [reply test/with_dummyserver/test_chunked_transfer.py:41]: I'd lean for "we support it but don't do this, so type annotation forbid this".

@sethmlarson thoughts?

Comment (sethmlarson) [reply test/with_dummyserver/test_chunked_transfer.py:41]: ++ on support with type annotations forbidding it.

File: test/with_dummyserver/test_chunked_transfer.py [hunk 2]
```diff
@@ -33,22 +36,30 @@ def socket_handler(listener):
 
         self._start_server(socket_handler)
 
-    def test_chunks(self):
+    @pytest.mark.parametrize(
```
Comment (sethmlarson) [inline test/with_dummyserver/test_chunked_transfer.py:39]: :muscle: Love this!

**Review by sethmlarson — APPROVED** [review]
LGTM, thanks for the revisions!

**Pull request merged**